File: pset6/vigenere/vigenere.py
import sys
from cs50 import get_string
if len(sys.argv) != 2:
    print("Usage: ./vigenere k")
    sys.exit(1)

key = sys.argv[1]
len_key = len(key)

if key.isalpha() == False:
    print("Usage: ./vigenere k")
    sys.exit(1)


pt = get_string("plaintext:")
ct = ""

n = len(pt)
j = 0
for i in range(n):
    k  = key[j % len_key]
    if k.islower():
        key_num = ord(k) - ord('a')
    else:
        key_num = ord(k) - ord('A')

    if pt[i].islower():
        cc = (ord(pt[i]) + key_num)
        if cc > ord('z'):
            c = chr(cc -26)
        else:
            c = chr(cc)
        j += 1
    elif pt[i].isupper():
        cc = (ord(pt[i]) + key_num)
        if cc > ord('Z'):
            c = chr(cc -26)
        else:
            c = chr(cc)
        j += 1
    else:
        c = pt[i]
    ct += c
    print(c)

print(f"ciphertext: {ct}")

File: pset2/vigenere/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./vigenere k");
        return 1;
    }

    string key = argv[1];
    int len_key = strlen(key);

    for(int i = 0; i < len_key; i++)
    {
        if(isalpha(key[i])== false)
        {
            printf("%c  sage: ./vigenere k!",key[i]);
            return 1;
        }
    }

    string pt,ct;
    pt = get_string("plaintext:");
    ct = pt;

    int j = 0;


    for(int i = 0, n = strlen(pt); i < n; i++)
    {
        int key_num;
        if (islower(key[(j % len_key)]))
        {
            key_num = key[(j % len_key)] - 'a';
        }
        else
        {
            key_num = key[(j % len_key)] - 'A';
        }

        if (islower(pt[i]))
        {

            if (pt[i] + key_num> 'z')
            {
                ct[i] = pt[i] + key_num -26;
            }
            else
            {
                ct[i] = pt[i] + key_num;
            }
            j++;
        }
        else if (isupper(pt[i]))
        {
            if (pt[i] + key_num  > 'Z')
            {
                ct[i] = pt[i] + key_num - 26;
            }
            else
            {
                ct[i] = pt[i] +key_num;
            }
            j++;
        }
        else
        {
            ct[i] = pt[i];
        }


    }

    printf("ciphertext: %s\n", ct);
}